lib/index.d.ts

```diff
@@ -4,22 +4,21 @@ import * as stream from 'stream'
 
 export = stringify
 
-declare function stringify(callback?: stringify.Callback): stringify.Stringifier
-declare function stringify(options?: stringify.Options, callback?: stringify.Callback): stringify.Stringifier
-declare function stringify(input: stringify.Input, callback?: stringify.Callback): stringify.Stringifier
-declare function stringify(input: stringify.Input, options?: stringify.Options, callback?: stringify.Callback): stringify.Stringifier
+declare function stringify(): stringify.Stringifier
+declare function stringify(callback: stringify.Callback): stringify.Stringifier
+declare function stringify(options: stringify.Options, callback: stringify.Callback): stringify.Stringifier
```

Comment:
Its fine to make `options` required, but `callback` here, and in every other signature, *must* be optional, to support the stream API.

Reply:
I took another look at the API and I agree with you: my function definition isn't good, mainly because of the _mixed API_ behaviour.
I reverted the function definitions, did some digging and did a minor fix.